fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/OCFLPersistentStorageSessionTest.java

```diff
@@ -591,4 +597,97 @@ public void getTriplesFromPreviousVersion() throws Exception {
+
+        // perform the create non-rdf source operation
+        session.persist(binOperation);
+
+        // commit to OCFL
+        session.commit();
+
+        // create a new session and verify the returned rdf stream.
+        final var newSession = createSession(index, objectSessionFactory);
+        final var result = IOUtils.toString(newSession.getBinaryContent(RESOURCE_ID, null), UTF_8);
+
+        assertEquals(BINARY_CONTENT, result);
+    }
+
+    @Test
+    public void getBinaryContentFailsIfAlreadyCommitted() throws Exception {
+        mockMappingAndIndex(mintOCFLObjectId(RESOURCE_ID), RESOURCE_ID, ROOT_OBJECT_ID, mapping);
+
+        // create the binary
+        final var binOperation = mockNonRdfSourceOperation(BINARY_CONTENT, USER_PRINCIPAL, RESOURCE_ID);
+
+        // perform the create non-rdf source operation
+        session.persist(binOperation);
+
+        // commit to OCFL
+        session.commit();
+
+        try {
+            session.getBinaryContent(RESOURCE_ID, null);
+            fail("Get must fail due to session having been committed");
+        } catch (final PersistentStorageException ex) {
+            // expected failure, handled with catch since the persist can throw same error
+        }
+    }
+
+    @Test
+    public void getBinaryContentVersion() throws Exception {
+        DefaultOCFLObjectSession.setGlobaDefaultCommitOption(NEW_VERSION);
+        // SEE getTriplesFromPreviousVersion
+        mockMappingAndIndex(mintOCFLObjectId(RESOURCE_ID), RESOURCE_ID, ROOT_OBJECT_ID, mapping);
+
+        // create the binary
+        final var binOperation = mockNonRdfSourceOperation(BINARY_CONTENT, USER_PRINCIPAL, RESOURCE_ID);
+        // perform the create non-rdf source operation
+        session.persist(binOperation);
+        // commit to OCFL
+        session.commit();
+
+        // create a new session and verify that the state is the same
+        final var newSession = createSession(index, objectSessionFactory);
+
+        final var versions = newSession.listVersions(RESOURCE_ID);
+        final var version1 = versions.get(versions.size() - 1);
+        assertEquals(BINARY_CONTENT,
+                IOUtils.toString(newSession.getBinaryContent(RESOURCE_ID, version1), UTF_8));
+    }
+
+    @Test
+    public void getBinaryContentUncommitted() throws Exception {
+        mockMappingAndIndex(mintOCFLObjectId(RESOURCE_ID), RESOURCE_ID, ROOT_OBJECT_ID, mapping);
+
+        // create the binary
+        final var binOperation = mockNonRdfSourceOperation(BINARY_CONTENT, USER_PRINCIPAL, RESOURCE_ID);
+
+        // perform the create non-rdf source operation
+        session.persist(binOperation);
+
+        // create a new session and verify the returned rdf stream.
+        final var result = IOUtils.toString(session.getBinaryContent(RESOURCE_ID, null), UTF_8);
+
+        assertEquals(BINARY_CONTENT, result);
+    }
+
+    private NonRdfSourceOperation mockNonRdfSourceOperation(final String content,
```

Comment:
Nit-pick, maybe the name should specify that you are mocking a Create NonRdfSource Operation. Eventually we might need one for Update as well.

Reply:
updated